lib/rules/custom-property-no-missing-var-function/index.js

```diff
@@ -32,12 +34,15 @@ const rule = (primary) => {
 			knownCustomProperties.add(atRule.params);
 		});
 
-		root.walkDecls(({ prop }) => {
-			if (isCustomProperty(prop)) knownCustomProperties.add(prop);
+		root.walkDecls(DASHED_IDENT_START, ({ prop }) => {
+			knownCustomProperties.add(prop);
 		});
 
 		root.walkDecls((decl) => {
 			const { value } = decl;
+
+			if (!DOUBLE_DASH.test(value)) return;
```

Comment:
[ask] How about using [`String.prototype.includes()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes) instead of a `RegExp`? No significant difference?

```js
const DOUBLE_DASH = "--";

// ...

if (!value.includes(DOUBLE_DASH)) return;
```

Reply:
I went a slightly different direction.

`if (!value.includes('--')) return;` should be fine because JavaScript has string interning, and strings don't need to be assigned to a `const` for performance.

-------

`root.walkDecls(/^--/, ({ prop }) => {` is more like the existing `root.walkAtRules(/^property$/i, (atRule) => {` above it.

These `RegExp`'s are only created once per stylelint invocation, so also no need to hoist these higher up.

----

thoughts?

Reply:
Make sense! 👍🏼